app.py

```diff
@@ -56,6 +56,24 @@ def experience():
         save_data('data/data.json', data)
         return jsonify({'id': new_id}), 201
     
+    if request.method == "PUT":
+        index = request.args.get("index")
+        if index is None:
+            return jsonify({"error": 'Index not provided'}), 400
+        if not index.isnumeric():
+            return jsonify({"error": "Index must be a number"}), 400
+        
+        index = int(index)
+        if not (0 < index <= len(data["experience"])):
```

Comment:
I think there's a small out of range issue here, might use < intead of <=

Reply:
No, that is correct, it should be <= because the ids are 1 indexed. If you remove it, and try it on pytest, it will prompt an error